### PR TITLE
Move crds to crds/

### PR DESCRIPTION
**Pull request by howardjohn**
This provides better support for helm3

For https://github.com/istio/istio/issues/22874

### CODE REVIEW

File: manifests/charts/base/files/gen-istio-cluster.yaml [hunk 1]
```diff
@@ -4969,7 +4952,26 @@ spec:
     storage: true
 ---
 
-
+---
+# Source: base/templates/serviceaccount.yaml
```
Comment (nrjpoddar) [inline manifests/charts/base/files/gen-istio-cluster.yaml:4956]: I thought these service accounts need to be renamed/moved

File: manifests/charts/base/files/gen-istio-cluster.yaml [hunk 2]
```diff
@@ -4980,7 +4982,7 @@ metadata:
   name: istiod-istio-system
   labels:
     app: istiod
-    release: istio-base
+    release: istio
```
Comment (nrjpoddar) [inline manifests/charts/base/files/gen-istio-cluster.yaml:4985]: Why this change?

**Review by nrjpoddar — COMMENTED** [review]
Minor comments, mostly LGTM.

File: manifests/charts/base/files/gen-istio-cluster.yaml (hunk 1 above)
Comment (nrjpoddar) [reply manifests/charts/base/files/gen-istio-cluster.yaml:4956]: I see that they will be moved in a later PR https://github.com/istio/istio/pull/23194

File: manifests/charts/base/files/gen-istio-cluster.yaml (hunk 2 above)
Comment (howardjohn) [reply manifests/charts/base/files/gen-istio-cluster.yaml:4985]: This is for kustomize - changing it to be the same as istioctl for consistent experience

File: operator/pkg/helm/helm.go [hunk 3]
```diff
@@ -148,6 +149,18 @@ func renderChart(namespace, values string, chrt *chart.Chart) (string, error) {
 			return "", err
 		}
 	}
+	for _, crdFile := range crdFiles {
+		f := string(crdFile.File.Data)
+		// add yaml separator if the rendered file doesn't have one at the end
```
Comment (nrjpoddar) [inline operator/pkg/helm/helm.go:154]: Just add the YAML separator, who cares?

**Review by nrjpoddar — APPROVED** [review]

File: operator/pkg/helm/helm.go (hunk 3 above)
Comment (howardjohn) [reply operator/pkg/helm/helm.go:154]: Not sure - just copied the code above :slightly_smiling_face:

File: Makefile.core.mk [hunk 4]
```diff
@@ -363,8 +363,8 @@ gen-check: gen check-clean-repo
 
 # Generate kustomize templates.
 gen-kustomize:
-	helm template -n istio-base manifests/charts/base > manifests/charts/base/files/gen-istio-cluster.yaml
-	helm template -n istio-base --namespace istio-system manifests/charts/istio-control/istio-discovery \
+	helm3 template istio --include-crds manifests/charts/base > manifests/charts/base/files/gen-istio-cluster.yaml
+	helm3 template istio --namespace istio-system manifests/charts/istio-control/istio-discovery \
```
Comment (costinm) [inline Makefile.core.mk:367]: I think that was a bug in the original - the base must be different ( or prune will go mad )

**Review by costinm — APPROVED** [review]

**Review by nmittler — APPROVED** [review]
